robota_core/repository.py

```diff
@@ -107,35 +127,41 @@ def _event_from_dict(self, event_data: dict):
 class Diff:
     """A representation of a git diff between two points in time for a single
     file in a git repository."""
-    def __init__(self, diff_info, diff_source: str):
+    def __init__(self, diff_info, diff_source: str, diff_url: str = None, page: int = 1):
         self.old_path: str
         self.new_path: str
         self.new_file: bool
         self.diff: str
+        self.url: str
 
         if diff_source == "gitlab":
-            self._diff_from_gitlab(diff_info)
+            self._diff_from_gitlab(diff_info, diff_url=diff_url, page=page)
```

Comment:
Oh - thanks.  That's useful to know.

Since the point of robota-core is to give a common interface to hosted Git projects, hiding differences like this between the providers, I think it would be better if we handled the paging entirely within this method and didn't expose the paging to callers.

You can ask GitLab to provide the full result, without paging.  I think it would be better to use that form of the API call, so that the caller gets the full result regardless of whether they are looking at commit diffs hosted in GitLab or GitHub, even at the risk of the result being too big for the memory in some cases.  Does that sound do-able?